mu/modes/circuitpython.py

```diff
@@ -236,6 +270,34 @@ def get_volume_name(disk_name):
                 self.connected = False
             return wd
 
+    def compatible_board(self, port):
+        """Use adafruit_board_toolkit to find out whether a board is running
+        CircuitPython. The toolkit sees if the CDC Interface name is appropriate.
+        """
+
+        pid = port.productIdentifier()
+        vid = port.vendorIdentifier()
+        manufacturer = port.manufacturer()
+        serial_number = port.serialNumber()
+        port_name = self.port_path(port.portName())
+
+        # Find all the CircuitPython REPL comports,
+        # and see if any of their device names match the one passed in.
+        for comport in circuitpython_serial.repl_comports():
+            if comport.device == port_name:
+                return Device(
+                    vid,
+                    pid,
+                    port_name,
+                    serial_number,
+                    manufacturer,
+                    self.name,
```

Comment:
I guess in this case the device will always be identified as "CircuitPython"?
I was hoping to go in the other direction and include more Adafruit board names in this mode to be able to see that in the UI (of course, always falling back to a default name for any device with the right VID, but unknown PID).

Reply:
That's right, so it's a bit of a functionality regression. But he current list is woefully incomplete, so most boards were identifying incorrectly as "CircuitPlayground".

I already thought about adding some board-name lookup capability to the `adafruit_board_toolkit`. Maybe it would be better to do that as a separate PR in order not to pile too much on this PR? I can work on that soon, I just didn't want to hold up this PR.

Reply:
> the current list is woefully incomplete, so most boards were identifying incorrectly as "CircuitPlayground".

Oh yeah, that's the main thing I wanted to do, populate that list with more board names, and ensure the `compatible_board()` method was able to first check for existing VID/PID pairs and then fallback if there is a VID/`None` entry.
(Sounds like a great use case for the new PEP 634 pattern matcher, too bad we need to support 3.5+ :)

I think we can try to match the name here with the PIDs in the original `valid_boards` list, that way we can keep it and that reduces the diff too.

Reply:
The current list is actually very out of date, and all Adafruit boards are already wildcarded to `"Adafruit CircuitPlayground"`. We actually support about 200 boards now.
https://github.com/mu-editor/mu/blob/f56bf64ff570d6268862b0396e79295031e6aedf/mu/modes/circuitpython.py#L43-L75

Reply:
Our answers crossed here. I'm suggesting instead that we remove the board names from the Mu codebase altogether. Instead, you'd call `adafruit_board_toolkit.circuitpython_board.name(vid, pid)` and it would give you the name, if it knew the name. Then Mu does not have to be updated constantly as new boards are added.

We will probably do some automation to keep `adafruit_board_toolkit` up to date in terms of board names. We can get the board names from circuitpython.org.

Reply:
> remove the board names
and the pid/vids